### PR TITLE
fix(NODE-5791): type error with $addToSet in bulkWrite

### PR DESCRIPTION
**Pull request by alenakhineika**
### Description
The `update` methods for `Collection` and `BulkWrite` with `$addToSet` should not show a _type_ error for collections with shema, collections extended from the `Document`/`any` type, and fields of `any` type.

#### What is changing?
- The type of `update` in `Collection.UpdateOneModel`/`Collection.UpdateManyModel`/`BulkWrite.updateOne`/`BulkWrite.updateMany` becomes `UpdateFilter<TSchema> | Document[]`.
- The `SetFields` type helper skips validation for fields of `any` type (provided by custom interface or index signatures).

##### Is there new documentation needed for these changes?
None

#### What is the motivation for this change?
[NODE-5647](https://jira.mongodb.org/browse/NODE-5647)
[NODE-4664](https://jira.mongodb.org/browse/NODE-4664)

### The type error is fixed for fields of `any` type in update commands.
Extending collection interfaces from `Document`/`any` type or specifying fields of `any` type doesn't cause a type error anymore.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/collection.ts [hunk 1]
```diff
@@ -340,12 +340,17 @@ export class Collection<TSchema extends Document = Document> {
...
    */
   async updateOne(
     filter: Filter<TSchema>,
-    update: UpdateFilter<TSchema> | Partial<TSchema>,
+    /**
+     * The value of update can be either:
+     * UpdateFilter<TSchema> - A document that contains update operator expressions,
+     * Document[] - an aggregation pipeline.
+     * */
+    update: UpdateFilter<TSchema> | Document[],
```
Comment (nbbeeken) [inline src/collection.ts:353]: A TSDoc comment on a parameter like this doesn't render in the docs, you can run `npm run build:docs` and `npm run preview:docs` to see for yourself

File: src/collection.ts [hunk 2]
```diff
@@ -340,12 +340,17 @@ export class Collection<TSchema extends Document = Document> {
    * Update a single document in a collection
    *
```
Comment (nbbeeken) [inline src/collection.ts:341]: I would add the below comment up here and format it a bit differently to get it to render nicely, you can use markdown syntax (ex. bullets and fixed-width):

```ts
  /**
   * Update a single document in a collection
   *
   * The value of `update` can be either:
   * - UpdateFilter<TSchema> - A document that contains update operator expressions,
   * - Document[] - an aggregation pipeline.
   *
   * @param filter - The filter used to select the document to update
   * @param update - The modifications to apply
   * @param options - Optional settings for the command
   */
```

**Review by nbbeeken — CHANGES_REQUESTED** [review]

File: src/collection.ts (hunk 1 above)
Comment (alenakhineika) [reply src/collection.ts]: TIL ✨

File: src/collection.ts (hunk 2 above)
Comment (alenakhineika) [reply src/collection.ts:341]: Updated with your suggestion.

**Review by nbbeeken — CHANGES_REQUESTED** [review]

**Review by nbbeeken — APPROVED** [review]

**Review by durran — APPROVED** [review]

**Pull request merged**